### PR TITLE
Bug/v1-187 | Fixed RoleRoute after reloading the page

### PR DESCRIPTION
**Pull request by PechurinaLena**
Fixed bug where no-founded page was displayed after reloading the page

### CODE REVIEW

File: frontend/src/components/Routes/RoleRoute.tsx [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-  children: React.ReactNode;
-  roles?: Role[];
-}
-
-const RoleRoute: React.FC<IRoleRouteProps> = ({ children, roles }) => {
+const RoleRoute: FC<IRoleRouteProps> = ({ children, roles }) => {
   const { data: userProfileResponse } = useGetProfile();
   const isAllowed = roles?.includes(userProfileResponse?.role as Role);
-  return isAllowed ? <>{children}</> : <Navigate to="/not-found" />;
+  const isPageReloaded = window.performance && window.performance.navigation.type === 1;
```
Comment (irina-sinitsyna) [inline frontend/src/components/Routes/RoleRoute.tsx:11]: please remove magic number

File: frontend/src/components/Routes/RoleRoute.tsx [hunk 2]
```diff
@@ -1,18 +1,16 @@
-import React from 'react';
+import { FC } from 'react';
 import { Navigate } from 'react-router-dom';
 
 import { useGetProfile } from 'api/profile';
 import { Role } from 'constants/menuRoles';
+import { IRoleRouteProps } from './types';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Routes/RoleRoute.tsx:6]: imports order

File: frontend/src/components/Routes/RoleRoute.tsx [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-}
-
-const RoleRoute: React.FC<IRoleRouteProps> = ({ children, roles }) => {
+const RoleRoute: FC<IRoleRouteProps> = ({ children, roles }) => {
   const { data: userProfileResponse } = useGetProfile();
   const isAllowed = roles?.includes(userProfileResponse?.role as Role);
-  return isAllowed ? <>{children}</> : <Navigate to="/not-found" />;
+  const isPageReloaded = window.performance && window.performance.navigation.type === 1;
+
+  return isAllowed || isPageReloaded ? <>{children}</> : <Navigate replace to="/not-found" />;
```
Comment (irina-sinitsyna) [inline frontend/src/components/Routes/RoleRoute.tsx:13]:
```suggestion
  return isAllowed || isPageReloaded ? <>{children}</> : <Navigate replace to={PATHS.notFound} />;
```

File: frontend/src/components/Routes/types.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { ReactNode } from 'react';
+import { Role } from 'constants/menuRoles';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Routes/types.ts:2]: imports order

**Review by irina-sinitsyna — CHANGES_REQUESTED** [review]

File: frontend/src/components/Routes/types.ts (hunk 4 above)
Comment (irina-sinitsyna) [reply frontend/src/components/Routes/types.ts:2]: also please create a folder types with index.ts inside instead of types.ts

**Review by ostapdans99 — APPROVED** [review]

**Review by irina-sinitsyna — APPROVED** [review]

**Pull request merged**